include/rocksdb/env.h

```diff
@@ -438,7 +443,31 @@ class WritableFile {
   }
   virtual ~WritableFile();
 
+  // Indicates if the class makes use of unbuffered I/O
+  virtual bool UseOSBuffer() const {
+    return true;
+  }
+
+  // This is needed when you want to allocate
+  // AlignedBuffer for use with file I/O classes
+  // Used for unbuffered file I/O when UseOSBuffer() returns false
+  virtual size_t GetRequiredBufferAlignment() const {
+    return 4 * 1024;
+  }
+
   virtual Status Append(const Slice& data) = 0;
+
+  // Positional write for unbuffered access default forward
+  // to simple append as most of the tests are buffered by default
+  virtual Status Append(const Slice& /* data */, uint64_t /* offset */) {
+    return Status::NotSupported();
+  }
+
+  // Truncate is necessary to trim the file to the correct size
+  // before closing. It is not always possible to keep track of the file
+  // size due to whole pages writes. The behavior is undefined if called
+  // with other writes to follow.
+  virtual Status Truncate(uint64_t size) = 0;
```

Comment:
Can you leave a default to return OK?
I want users who implement their own Env outside RocksDB to avoid code changes to compile.